### DISCUSSION TITLE
How does connaisseur validate its own image

### DISCUSSION THREAD

**Discussion by mathuvenkat** [sse-secure-systems/connaisseur#614] [Q&A] [answered]
Connaisseur comes with predefined validators for securesystemsengineering_official, however   i**mage: securesystemsengineering/connaisseur:v2.5.2** is what is defined in the deployment spec. Without the connaisseur webhook running how does it validate is own image. Pointers to code will help. Thanks!

**Comment by xopham** [comment, marked as answer]
@mathuvenkat upon initial rollout the image can only be validated manually via docker content trust. As you correctly hint, there is a _chicken or egg_ problem. However, once connaisseur is installed it will validate each subsequent image during upgrade via `helm upgrade` using the [preconfigured key](https://github.com/sse-secure-systems/connaisseur/blob/master/helm/values.yaml#L117-L122) by applying the [corresponding image policy](https://github.com/sse-secure-systems/connaisseur/blob/master/helm/values.yaml#L135-L138).

Hope that helps :slightly_smiling_face:

**Reply by mathuvenkat** [reply]
Thank you. So on first roll out to the cluster, if we are using helm operator to do so, the image cannot be validated, but subsequently on upgrades or values change it should be able to, from what I understand. 
What does validate manually via DCT mean - is it enabling docker client on the cluster to be checking if if the image is signed, but if we use containerd directly is there anyway to enforce this ?

**Reply by xopham** [reply]
Correct. Once installed you should even be able to inspect Connaisseur validation of itself if monitoring its logs during upgrade.
I am not aware that any direct integration with containerd exists. Manually validating would mean to do a docker pull with docker content trust active, to get a valid digest and compare that to the one deployed to production. That should ensure that the first installation was authentic. Unless Connaisseur is uninstalled later (or a breaking change occurs), all future releases should be verified.
